dvc/external_repo.py

```diff
@@ -69,6 +69,20 @@ def _external_repo(url=None, rev=None, cache_dir=None):
 
     repo = Repo(new_path)
     try:
+        # check if the URL is local and no default remote is present
+        # add default remote pointing to the original repo's cache location
+        if os.path.isdir(url):
+            rconfig = RemoteConfig(repo.config)
+            if not rconfig.remote_config_exists():
+                original_repo = Repo(url)
+                rconfig.add(
+                    "upstream",
```

Comment:
Would call it something like
```suggestion
                    "auto-generated",
```
or something else. `upstream` is too generic, it might conflict with existing sections named like that, so it is better to reflect "auto-generated" nature of this setting. Maybe there are better names though.